Добавление возврата со стирки.

Katetim32 opened the pull request:
Ваня решил полноценно включать в следующий спринт.

Katetim32 committed:
Добавила скрипт, маппинг и свойство в доменную модель.

Katetim32 committed:
Добавила кнопку во вьюшку возврата.

Katetim32 committed:
Добавила методы, обработала добавление.

Katetim32 committed:
Merge remote-tracking branch 'origin/master' into feature/I-1620

Katetim32 committed:
Добавление запятой после мержа

Katetim32 committed:
Добавление апдейта, проверок. Исправление ошибки null в клейме.

Katetim32 committed:
Исправление добавления количества

Katetim32 committed:
Добавила обработчик для удаления док-та возврата, исправила маппинг.

Katetim32 committed:
Обработала удаление строки возврата со стирки, чтобы менялся статус (+добавила этот новый статус в стирку).

Katetim32 committed:
Пока скрыла полностью кнопку.

Gankov commented on Workwear.Desktop/Domain/Stock.Documents/ReturnItem.cs:
```diff
@@ -153,10 +164,12 @@ public virtual DutyNormIssueOperation ReturnFromDutyNormOperation {
+				if(IssuedEmployeeOnOperation == null && IssuedDutyNormOnOperation == null && ServiceClaim != null)
+					return ReturnFrom.Claim;
 				throw new InvalidOperationException(
 					"Строка документа списания находится в поломанном состоянии. " +
```
Тут документ списания, а должен быть возврат.

Gankov commented on Workwear/Journal/ViewModels/ClothingService/ClaimsJournalViewModel.cs:
```diff
@@ -213,7 +213,7 @@ public class ClaimsJournalNode {
 		public uint ReferredTerminalId { get; set; }
 		public string Comment { get; set; }
-
+		public int Balance { get; set; }
```
Что это за поле? Что оно означает.

Gankov committed:
Merge remote-tracking branch 'origin/master' into feature/I-1620

# Conflicts:
#	Workwear.Desktop/Domain/Stock.Documents/Return.cs
#	Workwear.Desktop/Domain/Stock.Documents/ReturnItem.cs
#	Workwear.Desktop/Tools/BusinessLogicGlobalEventHandler.cs
#	Workwear.Sql/Scripts/2.10.1.sql
#	Workwear/ViewModels/Stock/ReturnViewModel.cs
#	Workwear/gtk-gui/gui.stetic

Gankov committed:
Удалил не нужный статус

SpaceTranquility committed:
Добавление сканером

Gankov committed:
Похоже пропустили и не почистили, интерфейс не нужен.

Gankov committed:
Так попробовал доделалть возврат из стирки.

Пока не тестил.

Gankov committed:
Доп пинформация для агентов, чтобы не забывали рекдактировать стетик

Gankov committed:
Merge pull request #802 from QualitySolution/feature/overnorm_add

Добавление сканером

Gankov committed:
Исправил удаление по тестам.

Gankov committed:
Merge remote-tracking branch 'origin/feature/I-1620' into feature/I-1620

Gankov committed:
Переделал журнал на использование хебернейта.

Gankov committed:
Подкорректировал документацию

Gankov committed:
Добавил фильтр овернормы по типу выдачи

Gankov committed:
исправил баг в модели из за чего она не зназачал тип выдачи.

+ Тесты на такую же ошибку в будуещм

Gankov committed:
Разрешил возвращать и простой выдачи тоже.

Gankov committed:
Добавил выбор конкретных штрих кодов, при выборе возврата со штрих кодами.

+ рефакторинг, вынес много запросов в репозиторий. Чтобы не перегружать и в будущем возможно меньше дублировать код.

Gankov committed:
Добавил отображение колонки со штрих кодами.

Gankov committed:
Добавилен возврат конкретных штрих кодов так же с дежурных норм и от сотрудников.

Gankov committed:
Запретил редактировать количество если возврат добавлялся по штрихкодам

Так же добавил эту проверку в валидаци.

Gankov committed:
Исправил валидацию, она не корректно проверяла тип списания.

Gankov committed:
Исправил интеграционный тест.

Перевел его с sqllite на контейнеры.

SpaceTranquility committed:
Некоректная операция возврата

SpaceTranquility committed:
Так же со списанием

SpaceTranquility committed:
Тесты на корректность пересчёта списания и возврата

Gankov committed:
при возврате и выдачи свехрнормы корректно заполняем операцию баркода, а так же зармеры.

Были проблемы.

Gankov committed:
Merge remote-tracking branch 'origin/feature/I-1620' into feature/I-1620

Gankov committed:
Короче запучался настраивать удаление чтобы оно работало корректно,

Мешаши вездесущие каскады на операциях, мне кажетя они вообще вредны, если нет прямого владения. Тут при таком количестве перекрестных ссылко прямого владения явно не наблюдаетртся, поэтму каскад олл точно нельзя. Написал специально тест, чтобы если ситуация поломается, это было сразу видно. Кейс трудно воспроизвести наврятли кто-то будет проверять в графике такое.

SpaceTranquility committed:
Добавление с разных операций, поддержжка множественного выбора из журнала

SpaceTranquility committed:
Добавил добавление сканером

Gankov committed:
Исправление работы других тестов.

Gankov committed:
Нашел 2 случая, с реальными багами сдела на них тесты.

Gankov committed:
Merge remote-tracking branch 'origin/feature/I-1620' into feature/I-1620

Gankov committed:
Исправление кейса в возвратом.

Gankov committed:
Пробую исправить причину падения одного из тестов.

SpaceTranquility committed:
Баг при работе с несохранённым документом

SpaceTranquility committed:
Merge remote-tracking branch 'origin/feature/I-1620' into feature/I-1620

Gankov committed:
исправление тестов в контейнерах.

Старый тип почему то не работал с новыми контейнерами.

Gankov committed:
Merge remote-tracking branch 'origin/feature/I-1620' into feature/I-1620

SpaceTranquility committed:
Комментарий  к строке

Gankov committed:
Исправлено отображение количества для всех печатных форм

Не показывалось дл ясверхнормы. и решение до этого откуда брать было не удачно.

Gankov committed:
Merge remote-tracking branch 'origin/feature/I-1620' into feature/I-1620

Gankov committed:
Увеличил время на тест, что бы хватало на запуск контейнера.

Gankov committed:
Блин 30 сек, не хватило.

Gankov committed:
Удалил не понятно на кой черт добавленное не нужное свойство.

Gankov committed:
Исправил комментарий на более корректный

А так же сделал модели обязательными, чтобы не было ложного представления о том что они опциональные.
